RecoTracker/MkFitCore/src/MaterialEffects.h

```diff
@@ -0,0 +1,441 @@
+          if (r < 55)
+            return 84;
+          else if (r < 80)
+            return 85;
+          else
+            return 86;
+        }
+        if (z < 174) {
+          if (r < 45)
+            return 87;
+          else
+            return 88;
+        }
+
+        if (z < 177.3) {
+          if (r < 55)
+            return 89;
+          else if (r < 80)
+            return 90;
+          else
+            return 91;
+        }
+        if (z < 181) {
+          if (r < 45)
+            return 92;
+          else
+            return 93;
+        }
+
+        if (z < 185) {
+          if (r < 55)
+            return 94;
+          else if (r < 80)
+            return 95;
+          else
+            return 96;
+        }
+        if (z < 188.5) {
+          if (r < 45)
+            return 97;
+          else
+            return 98;
+        }
+
+        if (z < 192) {
+          if (r < 55)
+            return 99;
+          else if (r < 80)
+            return 100;
+          else
+            return 101;
+        }
+        if (z < 195) {
+          if (r < 45)
+            return 102;
+          else
+            return 103;
+        }
+
+        if (z < 202) {
+          if (r < 55)
+            return 104;
+          else if (r < 80)
+            return 105;
+          else
+            return 106;
+        }
+        if (z < 206) {
+          if (r < 45)
+            return 107;
+          else
+            return 108;
+        }
+
+        if (z < 210) {
+          if (r < 55)
+            return 109;
+          else if (r < 80)
+            return 110;
+          else
+            return 111;
+        }
+        if (z < 212) {
+          if (r < 45)
+            return 112;
+          else
+            return 113;
+        }
+
+        if (z < 222) {
+          if (r < 55)
+            return 114;
+          else if (r < 80)
+            return 115;
+          else
+            return 116;
+        }
+        if (z < 224)
+          return 117;
+
+        if (z < 228) {
+          if (r < 55)
+            return 118;
+          else if (r < 80)
+            return 119;
+          else
+            return 120;
+        }
+        if (z < 232)
+          return 121;
+
+        if (z < 241) {
+          if (r < 55)
+            return 122;
+          else if (r < 80)
+            return 123;
+          else
+            return 124;
+        }
+        if (z < 245)
+          return 125;
+
+        if (z < 248) {
+          if (r < 55)
+            return 126;
+          else if (r < 80)
+            return 127;
+          else
+            return 128;
+        }
+        if (z < 252)
+          return 129;
+
+        if (z < 264) {
+          if (r < 80)
+            return 130;
+          else
+            return 131;
+        }
+        if (z < 267)
+          return 132;
+
+        if (z < 270) {
+          if (r < 80)
+            return 133;
+          else
+            return 134;
+        }
+        if (z < 280)
```

Comment:
These do not repeat. So, the usual rule to define named constants does not really apply.

The documentation is available in trackreco/mkFit#278

I've added comments inline in Round 5 follow-up PR trackreco/cmssw#80 commit https://github.com/trackreco/cmssw/pull/80/commits/bc895c8ad17635da313fb0ec7dd47aef0ec8ab5f

Reply:
Since when it's fine to use "magic numbers" instead of named constants simply because they are only used once ?
In our last round of PRs we received plenty of requests to document all possible constants used in the code.

In the CMSSW coding rules, [**7**](https://cms-sw.github.io/cms_coding_rules.html#7--design-and-coding-guidelines-1)**.6** says

> Do not use magic numbers. Deﬁne constants using `enum class` or `constexpr` [...].

and it doesn't make any mention of "if they are used more than once".

Reply:
my comment related to the named constants to be only for repeated literals was based on my past comments.
Just in case it was a current induced memory, I checked my comments to PRs in 2021 (I searched for "constant" in the body) and all were referring to repeated literals.